### DISCUSSION TITLE
Large webapp state machines architecture communcation

### DISCUSSION THREAD

**Discussion by tomeraz-ann** [statelyai/xstate#3777] [Q&A]
I'm designing an online editor architecture using vue.js and xstate. I want to use many state machines, at different levels of the webapp. Ideally have small number of states for each machine to keep things simple.

How do I communicate between state machines that are independent?

Some of the machines have parent-child relationships, and can communicate using the actor model. But some machines have no patent-child relationship.

For example having the autosave machine react to an edit machine, or a user notification machine react to a data fetching machine.

Do I have to design them in the same parent-child hierarchy in order to send messages between actors?

**Comment by davidkpiano** [comment]
We're still planning a receptionist pattern for v5 that would handle this. I'll be opening a PR soon for this.

**Reply by tomeraz-ann** [reply]
Thanks 🙏🏼
Until v5 is out, do you recommend working with pub/sub along the machines?

**Reply by davidkpiano** [reply]
You can look at 3rd-party tools like XSystem: https://github.com/christoph-fricke/xsystem
